lib/builders/graviton-builder.ts

```diff
@@ -0,0 +1,172 @@
+        );
+    }
+
+    public addMetricsServerAddOn(props?: addons.MetricsServerAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.MetricsServerAddOn(props)
+        );
+    }
+
+    public addPrometheusNodeExporterAddOn(props?: addons.PrometheusNodeExporterAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.PrometheusNodeExporterAddOn(props)
+        );
+    }
+
+    public addExternalsSecretsAddOn(props?: addons.ExternalsSecretsAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.ExternalsSecretsAddOn(props)
+        );
+    }
+
+    public addSecretsStoreAddOn(props?: addons.SecretsStoreAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.SecretsStoreAddOn(props)
+        );
+    }
+
+    public addCalicoOperatorAddOn(props?: addons.CalicoOperatorAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.CalicoOperatorAddOn(props)
+        );
+    }
+
+    public addCertManagerAddOn(props?: addons.CertManagerAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.CertManagerAddOn(props)
+        );
+    }
+
+    public addAdotCollectorAddOn(props?: addons.AdotCollectorAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.AdotCollectorAddOn(props)
+        );
+    }
+
+    public addAmpAddOn(props: addons.AmpAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.AmpAddOn(props)
+        );
+    }
+
+    public addCloudWatchLogsAddOn(props: addons.CloudWatchLogsAddonProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.CloudWatchLogsAddon(props)
+        );
+    }
+
+    public addEfsCsiDriverAddOn(props?: addons.EfsCsiDriverProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.EfsCsiDriverAddOn(props)
+        );
+    } 
+
+    public addFluxCDAddOn(props?: addons.FluxCDAddOnProps) : GravitonBuilder {  
+        return this.addOns(
+            new addons.FluxCDAddOn(props)
+        );
+    }  
+
+    public addGrafanaOperatorAddOn(props?: addons.GrafanaOperatorAddonProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.GrafanaOperatorAddon(props)
+        );
+    }  
+
+    public addXrayAdotAddOn(props?: addons.XrayAdotAddOnProps) : GravitonBuilder {
+        return this.addOns(
+            new addons.XrayAdotAddOn(props)
+        );
+    }
+
+    public static builder(options: GravitonOptions): GravitonBuilder {
+        const builder = new GravitonBuilder();
+
+        builder
+            .clusterProvider(
+                new MngClusterProvider({
+                    version: options.kubernetesVersion,
+                    instanceTypes: [new ec2.InstanceType(`${options.instanceClass}.${options.instanceSize}`)],
+                    amiType: eks.NodegroupAmiType.AL2_ARM_64,
+                    desiredSize: 3,
```

Comment:
If we are to move it beyond just a demo, these should be configurable. We need to add something like `withNodeGroupOptions(options: Partial<ManagedNodeGroupProps>)` to the builder to allow any kind of an override. The approach is similar to the one I recommended to Bhavye on his builder.

Reply:
Why `desiredSize, min, max` are left hardcoded and not leveraging the potentially passed value from the options?

Reply:
My recommendation is to use default MngClusterProviderProps and overwrite them with the supplied. That will be easier to implement and provides the best flexibility. In the current implementation it is more of a demo rather than a production ready builder.